content/spin/v3/community-mqtt-trigger.md

```diff
@@ -0,0 +1,170 @@
+title = "The Spin MQTT Trigger (Community)"
+template = "spin_main"
+date = "2025-01-09T00:00:01Z"
+enable_shortcodes = true
+[extra]
+url = "https://github.com/fermyon/developer/blob/main/content/spin/v3/community-mqtt-trigger.md"
+
+---
+
+- [Installing the MQTT Trigger](#installing-the-mqtt-trigger)
+  - [Installing the MQTT Trigger Plugin](#installing-the-mqtt-trigger-plugin)
+  - [Installing the MQTT Trigger Template](#installing-the-mqtt-trigger-template)
+- [Creating the Application](#creating-the-application)
+  - [Configuring the MQTT Connection](#configuring-the-mqtt-connection)
+  - [Inspecting the Source Code](#inspecting-the-source-code)
+- [Building and Running the Application](#building-and-running-the-application)
+- [Additional Spin Triggers](#additional-spin-triggers)
+  - [Common Triggers](#common-triggers)
+  - [Community Triggers](#community-triggers)
+
+
+Spin has experimental support for creating and running components for MQTT messages. Please note that there are only working MQTT Trigger app samples written in [Rust](https://github.com/spinkube/spin-trigger-mqtt/tree/main/examples/mqtt-app) at present.
+
+> Please note: You can not `spin deploy` an application to Fermyon Cloud if it uses `mqtt` because non-HTTP triggers are not supported in Fermyon Cloud.
+
+Let's look at how the [experimental MQTT trigger for Spin](https://github.com/spinkube/spin-trigger-mqtt) allows you to deploy an application that runs for every message received via MQTT. A MQTT trigger maps a configurable MQTT service to a specific component. For example:
+
+<!-- @nocpy -->
+
+```toml
+[application.trigger.mqtt]
+address = "mqtt://localhost:1883"
+username = "user"
+password = "password"
+keep_alive_interval = "30"
+
+[[trigger.mqtt]]
+id = "mqtt-01"
+component = "hello"
+topic = "topic-one"
+qos = "1"
+```
+
+> Note: A single Spin App could consist of multiple MQTT triggers invoking different components for individual topics, as shown in the next snippet
+
+```toml
+[application.trigger.mqtt]
+address = "mqtt://localhost:1883"
+username = "user"
+password = "password"
+keep_alive_interval = "30"
+
+[[trigger.mqtt]]
+id = "mqtt-01"
+component = "hello-one"
+topic = "topic-one"
+qos = "1"
+
+[[trigger.mqtt]]
+id = "mqtt-02"
+component = "hello-two"
+topic = "topic-two"
+qos = "1"
+```
+
+In the snippet above, you can see that different components are instantiated and invoked for messages appearing in topics `topic-one` and `topic-two`.
+
+
+## Installing the MQTT Trigger
+
+### Installing the MQTT Trigger Plugin
+
+First, we install the plugin:
+
+```bash
+spin plugin install --url https://github.com/spinkube/spin-trigger-mqtt/releases/download/canary/trigger-mqtt.json --yes
```

Comment:
I thought we had stable versions in the plugins registry - is that not the case?  I'm wary of pointing people to canary builds.